kensu/matplotlib/extractor.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from hashlib import sha256
+
+import matplotlib as plt
+
+
+import kensu
+from kensu.client import *
+from kensu.utils.dsl.extractors import ExtractorSupport, get_or_set_rand_location
+from kensu.utils.helpers import singleton, save_stats_json, to_datasource
+from kensu.utils.kensu_provider import KensuProvider
+
+@singleton
+class PlotSupport(ExtractorSupport):  # should extends some KensuSupport class
+
+    def is_supporting(self, fig):
+        return isinstance(fig,  plt.figure.Figure)
+
+    def is_machine_learning(self, df):
+        return False
+
+    # return list of FieldDef
+    def extract_schema_fields(self, fig):
+        k=KensuProvider().instance()
+        fields = []
+        for ax in fig.axes:
+            for df in ax.inheritance:
+                fields_ax_df = k.extractors.extract_schema_fields(df)
+                fields = fields + fields_ax_df
+        return fields
+
+    def tk(self, k, k1):
+        return str(k) + '.' + k1
+
+    # return dict of doubles (stats)
+    def extract_stats(self, fig):
+        kensu = KensuProvider().instance()
+        stats = {}
+        ax_num = 0
+        for ax in fig.axes:
+            source_num = 0
+            for df in ax.inheritance:
+                inherited_stats = {'Ax'+str(ax_num)+'.Source'+str(source_num)+'.'+key : value for key,value in kensu.extractors.extract_stats(df).items()}
```

Comment:
we could also (later?) cover the case when axes are named! so to report axe names instead of number ids.

Reply:
added a #TODO